packages/components/src/entities/SampleHeader.tsx

```diff
@@ -0,0 +1,318 @@
+import { PicklistCreationMenuItem } from '../internal/components/picklist/PicklistCreationMenuItem';
+import { DisableableMenuItem } from '../internal/components/samples/DisableableMenuItem';
+import { SAMPLE_TIMELINE_AUDIT_QUERY } from '../internal/components/auditlog/constants';
+import { EntityDeleteModal } from './EntityDeleteModal';
+import { SampleTypeDataType } from '../internal/components/entities/constants';
+import {
+    LabelPrintingProviderProps,
+    withLabelPrintingContext
+} from '../internal/components/labels/LabelPrintingContextProvider';
+import { useServerContext } from '../internal/components/base/ServerContext';
+import { PrintLabelsModal } from '../internal/components/labels/PrintLabelsModal';
+import { CreateSamplesSubMenu } from './CreateSamplesSubMenu';
+import { AssayImportSubMenuItem } from './AssayImportSubMenuItem';
+
+interface StorageMenuProps {
+    sampleModel: QueryModel;
+    onUpdate?: (skipChangeCount?: boolean) => any;
+}
+
+interface HeaderProps {
+    assayProviderType?: string;
+    entityDataType?: EntityDataType;
+    onUpdate: () => void;
+    sampleModel: QueryModel;
+    showDescription?: boolean;
+    lastUpdatedCount?: number; // TODO Is this still needed? Probably added so the header would rerender after updates in the page
+    hasActiveJob?: boolean;
+    iconSrc?: string;
+    jobCreationHref?: string;
+    navigate: (url: string | AppURL) => void;
+    sampleContainer?: Container;
+    user?: User;
+    title?: string;
+    subtitle?: ReactNode;
+    isCrossFolder?: boolean;
+    StorageMenu?: ComponentType<StorageMenuProps>
+}
+
+export type Props = HeaderProps & LabelPrintingProviderProps;
+
+// exported for jest testing purposes only
+export const SampleHeaderImpl: FC<Props> = memo(props => {
+    const {
+        assayProviderType,
+        canPrintLabels,
+        entityDataType,
+        iconSrc = 'samples',
+        hasActiveJob,
+        labelTemplate,
+        navigate,
+        onUpdate,
+        printServiceUrl,
+        sampleContainer,
+        sampleModel,
+        showDescription,
+        isCrossFolder,
+        jobCreationHref,
+        title,
+        subtitle,
+        StorageMenu,
+    } = props;
+    const { queryInfo } = sampleModel;
+    const { createNotification } = useNotificationsContext();
+    const [canDelete, setCanDelete] = useState<boolean>(false);
+    const [error, setError] = useState<boolean>(false);
+    const [showConfirmDelete, setShowConfirmDelete] = useState<boolean>(false);
+    const [showPrintDialog, setShowPrintDialog] = useState<boolean>(false);
+    const sampleId = useMemo(() => sampleModel.getRowValue('RowId'), [sampleModel]);
+    const sampleIds = useMemo(() => [sampleId], [sampleId]);
+    const { user } = useServerContext();
+
+    useEffect((): void => {
+        (async () => {
+            try {
+                const confirmationData = await getSampleOperationConfirmationData(SampleOperation.Delete, undefined, sampleIds);
```

Comment:
the LKB implementation previously had an if statement to check for user perm and `isSampleOperationPermitted` before making this API call. Is that no longer needed here?

Here is what it was in LKB:
```
        if (
            user.hasDeletePermission() &&
            isSampleOperationPermitted(getSampleStatusType(sampleModel.getRow()), SampleOperation.Delete)
        )
```